src/taco-utils/telemetryHelper.ts

```diff
@@ -246,6 +246,38 @@ module TacoUtility {
                 () => generator.send()); // After
         }
 
+        /**
+         * Creates an object that can be passed to external modules to report telemetry events.
+         * @param componentName - a name to prefix on the telemetry event names received from the external module.
+         * @param baseProps - a collection of properties to include with all events.
+         * @param errorHandler - an optional function to wrap or modify any errors (wrap in a TacoError, for example).
+         * @returns {{sendTelemetry: (function(string, ITelemetryProperties, Error=): void)}}
+         */
+        public static getExternalTelemetryObject(componentName: string, baseProps: Telemetry.ITelemetryProperties, errorHandler?: (error: Error) => Error) : Object {
```

Comment:
I see interfaces as being especially useful when they are at the boundary between modules, as a form of documentation. You've basically defined the type in the JSDoc above, may as well make it formal and let the type system know about it too.